tests/PageObjects/Locator.py

```diff
@@ -0,0 +1,346 @@
+from __future__ import annotations
+import time
+from Actions import Action, Actions
+from selenium.common.exceptions import NoSuchElementException
+
+"""
+Locator object which will allow actions to be performed on an element while
+also ensuring the element requirements have been tried if element is not found
+the first time. Raises NoSuchElementException is attempt to locate fails
+"""
+
+
+class Locator(Actions):
+    def __init__(self, driver, loc, loc_type, descr, multiple, *reqs):
+        self.driver = driver
+        self.locator = loc  # Valid loc types: id, class, css, xpath
+        self.locator_type = loc_type
+        self.description = descr
+        self.is_multiple = multiple
+        self.requirements = reqs
+        self.is_child = False
+        self.element = None
+        self.__VERBOSE__ = True
+        self.__reset_defaults__()
+
+    """--------------- Helper Functions ---------------"""
+
+    def __reset_defaults__(self):
+        # Action Modifier, will modify how action is performed if set to true
+        self.__TRY__ = False  # Quit looking for element after first try
+        # Locator must be available to perform action
+        self.__REQUIRED_STATE__ = "available"
+
+    def __valid_state__(self):
+        state = self.__REQUIRED_STATE__
+        if state == "available":
+            return self.available()
+        if state == "visible":
+            return self.visible()
+        if state == "enabled":
+            return self.enabled()
+        if state == "selected":
+            return self.selected()
+
+    # Will perform requirement actions to prepare for getting locator
+    def __prepare__(self):
+        if self.requirements == (None,) or len(self.requirements) == 0:
+            return False
+
+        for req in self.requirements:
+            if self.__VERBOSE__:
+                print("Performing preparation steps...")
+            if type(req).__name__ == "Locator":
+                if self.__VERBOSE__:
+                    req.click()
+                else:
+                    req.silent().click()
+            elif type(req).__name__ == "Action":
+                req.perform(self.__VERBOSE__)
+            elif self.__VERBOSE__:
+                if self.description != "":
+                    print("Unkown requirement in locator: {}".format(
```

Comment:
do we need to return False here?

Reply:
Good catch, I've added a return false statement to handle when an unknown requirement is passed to a locator.